### PR TITLE
Debugger: show correct value for non-temporary names

### PR DESCRIPTION
**Pull request by gw000**
Automatically map user variables to temporary names.

Screen Shot 2021-07-23 at 6 22 21 PM

PR checklist:

- [x] Added changelog entry.

### CODE REVIEW

File: polar-core/src/debugger.rs [hunk 1]
```diff
@@ -292,12 +293,20 @@ impl Debugger {
                 if parts.len() > 1 {
                     let vars: Vec<Binding> = parts[1..]
                         .iter()
-                        .map(|var| {
-                            let var = Symbol::new(var);
-                            let value = vm.bindings(true).get(&var).cloned().unwrap_or_else(|| {
-                                Term::new_temporary(Value::Variable(Symbol::new("<unbound>")))
-                            });
-                            Binding(var, value)
+                        .map(|nom| {
```
Comment (saolsen) [inline polar-core/src/debugger.rs:296]: What is `nom` here?

File: polar-core/src/kb.rs [hunk 2]
```diff
@@ -45,16 +45,18 @@ impl KnowledgeBase {
         self.id_counter.clone()
     }
 
+    pub fn temp_prefix(nom: &str) -> String {
```
Comment (saolsen) [inline polar-core/src/kb.rs:48]: Could we maybe just add a comment explaining what this is for?

File: polar-core/src/debugger.rs [hunk 3]
```diff
@@ -292,12 +293,20 @@ impl Debugger {
...
+                            let var = Symbol::new(nom);
+                            let bindings = vm.bindings(true);
+                            bindings.get(&var).cloned().map_or_else(|| {
+                                let pref = KnowledgeBase::temp_prefix(nom);
```
Comment (saolsen) [inline polar-core/src/debugger.rs:300]: Lengthening this to `prefix` would help me too.

**Review by saolsen — APPROVED** [review]
Looks good, I'm excited to use this it would have come in handy last week for me.
I had a little trouble following how it worked, I think maybe a few comments and more descriptive variable names might have helped. Made a few suggestions.

File: polar-core/src/kb.rs (hunk 2 above)
Comment (gw000) [reply polar-core/src/kb.rs]: 👍

File: polar-core/src/debugger.rs (hunk 3 above)
Comment (gw000) [reply polar-core/src/debugger.rs]: 👍

File: polar-core/src/debugger.rs (hunk 1 above)
Comment (gw000) [reply polar-core/src/debugger.rs]: Just a synonym for name, changed it.

**Pull request merged**